plugins/cuda/cuda_plugin.c

```diff
@@ -145,33 +150,41 @@ static int launch_cuda_checkpoint(const char **args, char *buf, int buf_size)
 		}
 
 		int status;
-		if (waitpid(child_pid, &status, 0) == -1 || !WIFEXITED(status)) {
-			pr_err("cuda-checkpoint exited improperly, couldn't complete operation\n");
-			close(fd[READ]);
-			return -1;
-		}
+		int exit_code = waitpid(child_pid, &status, 0);
+		if (exit_code == -1 || !WIFEXITED(status))
```

Comment:
1. we use the `exit_code` for the very specific pattern, so pls choose another name for this variable. 
2. you print error here, but you don't change exit_code, so this function return 0 that doesn't look tight.

Reply:
Thank you Andrei! I have updated the pull request with these changes.